internals/overlord/state/state.go

```diff
@@ -293,6 +295,22 @@ func (s *State) Unlock() {
 			s.modified = false
 			return
 		}
+
+		// Let the user know what's going on if the backend writes failed.
+		if _, ok := err.(*os.PathError); ok {
+			logger.Noticef("backend checkpoint failed: %v", err)
+		}
+
+		// Channel to listen for interrupt signals.
+		interrupt := make(chan os.Signal, 1)
+		signal.Notify(interrupt, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
This has a couple of related issues:

- You're calling `signal.Notify` once every loop. Only needs to be done once.
- In any case, we don't want this here (`signal.Notify` is global state polluting low-level code). It'd be much cleaner if we gave state a channel that was closed by a new `st.Stop()` method, and this was a `select` that waited on the stop channel or the timer. We already listen for signals in `cmd_run.go` at the top level, and that's probably where that should stay.

Still, it's a bit tricky to know when to call `st.Stop()`, because `state.Unlock` is called in the setup phase, when `cmd_run.go` is not yet waiting for interrupt signals. I wonder if we have a `state.InitDone()` that is called when we're ready; before that it doesn't do retries, after that it does (and waits on the close channel).

Or maybe we don't need to retry at all here? I'm not sure why we retry the writing when it's local disk. Surely if it would fail once it'll fail again? I'd like to get @hpidcock's input on this.

Reply:
Yeah, after discussion with Harry, he pointed out this isn't what we want to do anyway -- we want the "atomic commit" to try as hard as possible and *not* be interruptible. If they really want to kill the process they can `SIGKILL` it hard.

So I think we should leave this as is, but keep the `IsWriteable` check (which will catch most problems much earlier) and keep the additional log.